### ISSUE TITLE
Plugin Admin replaces the user-defined template with the default template

### ISSUE THREAD

**Issue by petira** [getgrav/grav-plugin-admin#1701]
Plugin Admin replaces the user-defined template (which is not in the root folder `'active_theme\templates'`) with the default template. For example: `formdata.md` => `default.md`. Of course, the template can be set in the frontmatter, but it is not ideal.

This is partly related to #1689.

**Comment by ricardo118** [comment]
because the admin plugin reads only from the `templates` folder root on the active theme. Your best option if you want to have the file elsewhere is to create a template in the root with the correct name that includes another in a remote/different location

**Comment by petira** [comment]
Yes, it's how you write. However, I think it would be better to leave the Plugin Admin user-defined template without replacing it with the default template.

Instead of (in Pages):
**Thank you!** (based on `formdata.md`)
`/contact/thankyou => default`
specify:
`/contact/thankyou => [user-defined]`

And in `Pages > Advanced > Settings > Page Template`, will be selected template `[User Defined]` (without further specification) as active to keep it when changes are saved. Only if a user chooses a different tempate (Default, Form, Item, etc.) would it change.

Creating an imaginary template in the root folder is not ideal. It is better to define your own template in the page header.

**Comment by rhukster** [comment]
Like @ricardo118 says, there's no way to for Grav to know about other templates other than ones at the root of `yourtemplate/templates`.  However, you can use use an event in your plugin/theme to add extra templates (you need to also add a reference in the getSubscribedEvents method:

```
    public function onGetPageTemplates(Event $event)
    {
        /** @var Types $types */
        $types = $event->types;
        $types->register('bar');

        $types->scanTemplates('plugins://testing/templates');
    }
```

Here you can see I manually added a `bar` template, and also told it to scan a directory to get some extra template names.

More info here: https://learn.getgrav.org/16/admin-panel/extending#adding-a-custom-page-blueprint-to-a-theme-plugin

**Comment by petira** [comment]
I understand everything. But it is something else. If the page being edited is using a template other than that specified in `activetheme/templates`, **it should not be replaced by the default, but retained unless the user explicitly makes the change.**

The reasons may be different. For example, given formdata.md, whose sole purpose is to display the message after the form is submitted. There is no reason to be displayed in the template list. But it displays the information in the header (title) and in the content of the page. In this case, the solution is easy - the template can be default.md and the header is rewritten to formdata.md.

But it may happen that the user changes the theme to another that does not have certain templates. For example, I will have a Quark theme based blog and change to a different theme. I will edit posts based on `item.md` and when saved, the template will change to `default.md` because `item.md` does not know. Then I decide to use Quark theme again and the edited posts will be rendered by `default.md`.
